paasta_tools/flink_tools.py

```diff
@@ -166,6 +187,64 @@ def _dashboard_get(cr_name: str, cluster: str, path: str) -> str:
     return response.text
 
 
+def _filter_for_endpoint(json_response: Any, endpoint: str) -> Mapping[str, Any]:
+    """
+    Filter json response to include only a subset of fields.
+    """
+    if endpoint == "config":
+        return {
+            key: value for (key, value) in json_response.items() if key in CONFIG_KEYS
+        }
+    if endpoint == "overview":
+        return {
+            key: value for (key, value) in json_response.items() if key in OVERVIEW_KEYS
+        }
+    if endpoint == "jobs":
+        return json_response
+    if endpoint.startswith("jobs"):
+        return {
+            key: value
+            for (key, value) in json_response.items()
+            if key in JOB_DETAILS_KEYS
+        }
+    return json_response
+
+
+def _get_jm_rest_api_base_url(cr: Mapping[str, Any]) -> str:
+    metadata = cr["metadata"]
+    cluster = metadata["labels"][paasta_prefixed("cluster")]
+    base_url = get_flink_ingress_url_root(cluster)
+
+    # dashboard_url = http://flink-jobmanager-host:port/paasta-service-cr-name
+    dashboard_url = metadata["annotations"]["flink.yelp.com/dashboard_url"]
+    service_cr_name = dashboard_url[dashboard_url.rfind("/") + 1 :]
+
+    return base_url + service_cr_name
+
+
+def curl_flink_endpoint(cr_id: Mapping[str, str], endpoint: str) -> Mapping[str, Any]:
+    try:
+        cr = get_cr(settings.kubernetes_client, cr_id)
+        if cr is None:
+            raise ValueError(f"failed to get CR for id: {cr_id}")
+        base_url = _get_jm_rest_api_base_url(cr)
+        url = f"{base_url}/{endpoint}"
+        response = requests.get(url, timeout=FLINK_DASHBOARD_TIMEOUT_SECONDS)
+        if response.ok:
+            return _filter_for_endpoint(response.json(), endpoint)
+        return {}
```

Comment:
will the error we presumably encountered if we hit this case be exposed in some other way?

Reply:
I agree it's a good idea to propagate any error client-side, not just an empty body. I'll look around and see what we do in other handlers to try to reproduce it.